lib/tasks/petitions.rake

```diff
@@ -8,4 +8,22 @@ namespace :petitions do
 
     puts "Version: #{version_id} scheduled for publication"
   end
+
+  desc "Fetches petitions informations from the api and store it on the cache"
+  task fetch_info: :environment do |_, args|
+
+    petition_service = PetitionService.new
+
+    PetitionPlugin::Detail.all.find_in_batches do |batch|
+      batch.each do |detail|
+        begin
+          petition_service.fetch_petition_info detail.id, fresh: true
+          puts "Fetched information for petition #{detail.id}"
```

Comment:
But this will mainly be used as a cron job. We won't have any log this way. You could log both to `Rails.logger` and `stdout`.